rocAL/rocAL/include/meta_data/meta_data.h

```diff
@@ -81,70 +90,143 @@ typedef struct
     RotationBatch rotation_batch;
 }JointsDataBatch;
 
+typedef struct MetaDataInfo {
+public:
+    uint img_id = -1;
+    std::string img_name = "";
+    ImgSize img_size = {};
+} MetaDataInfo;
+
 struct MetaData
 {
-    int& get_label() { return _label_id; }
-    BoundingBoxCords& get_bb_cords() { return _bb_cords; }
-    BoundingBoxCords_xcycwh& get_bb_cords_xcycwh() { return _bb_cords_xcycwh; }
-    BoundingBoxLabels& get_bb_labels() { return _bb_label_ids; }
-    void set_bb_labels(BoundingBoxLabels bb_label_ids) {_bb_label_ids = std::move(bb_label_ids); }
-    ImgSize& get_img_size() { return _img_size; }
-    const JointsData& get_joints_data(){ return _joints_data; }
+    virtual void get_labels(std::vector<int>** label_ids) {}
```

Comment:
All the methods in base class should throw and exception (Not implemented) instead of doing nothing